hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/metadata/WitnessedContainerMetadataStoreImpl.java

```diff
@@ -0,0 +1,78 @@
+package org.apache.hadoop.ozone.container.metadata;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+import org.apache.hadoop.hdds.conf.ConfigurationSource;
+import org.apache.hadoop.hdds.utils.db.DBStore;
+import org.apache.hadoop.hdds.utils.db.DBStoreBuilder;
+import org.apache.hadoop.hdds.utils.db.Table;
+import org.apache.hadoop.hdds.utils.db.managed.ManagedDBOptions;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+/**
+ * Class for interacting with database in the master volume of a datanode.
+ */
+public final class WitnessedContainerMetadataStoreImpl extends AbstractRDBStore<WitnessedContainerDBDefinition>
+    implements WitnessedContainerMetadataStore {
+
+  private Table<Long, String> containerIdsTable;
+  private static final ConcurrentMap<String, WitnessedContainerMetadataStore> INSTANCES =
+      new ConcurrentHashMap<>();
```

Comment:
@swamirishi , It seems that the size of this map is at most 1.  Is it the case?

Reply:
Ping @swamirishi

Reply:
yeah you are right

Reply:
If you see the initial set of commits this was originally a singleton. But there were too many existing test case failures where all the tests run in the same jvm. So I had to change it to a concurrent map. The PR size increased a lot, so just took a short cut to get one instance. We can make the initial size of the the hash table to be just one. I guess that should bring in some optimization.

Reply:
If it has at most one entry, we should lazily initialize it or use AtomicReference.